extensions/eda/plugins/event_filter/normalize_keys.py

```diff
@@ -0,0 +1,50 @@
+"""
+normalize_keys.py:
+    An event filter that changes keys that contain non alpha numeric or
+    underscore to undersocres.
+    For instance, the key X-Y becomes the new key X_Y.
+    If there are consecutive non alpa numeric or under score, they would
+    be coalesced into a single underscore
+    For instance the key X/?Y becomes X_Y instead of X__Y
+
+    If there is a existing key with the normalized name, you can pass
+    in the overwrite argument to overwrite it with the new key
+Arguments:
+    * overwrite: Overwrite the values if there is a collision with a new key.
+
+Usage in a rulebook, a filter is usually attached to a source in the rulebook:
+    ....
+    sources:
+      - my_source:
+         ....
+        filters:
+          ansible.eda.normalize_keys:
+"""
+
+import multiprocessing as mp
+import re
+
+
+def main(event, overwrite=True):
+    logger = mp.get_logger()
+    logger.info("normalize_keys")
+    q = []
+    q.append(event)
+    regex = re.compile("[^0-9a-zA-Z_]+")
+    while q:
+        o = q.pop()
+        if isinstance(o, dict):
```

Comment:
We are skipping lists of dicts, don't?

Reply:
@Alex-Izquierdo A single event is always a dict its never an array.

Reply:
A single event can contain a list of dicts. 
```{ "name": "alex", "phones": [{"home": 777777}, {"work": 5555555}]}```